rotkehlchen/tests/api/test_ethereum_transactions.py

```diff
@@ -307,6 +307,28 @@ def mock_etherscan_get(url, *args, **kwargs):  # pylint: disable=unused-argument
         assert len(events) == 1
         assert events[0].identifier not in event_ids
 
+    # check that passing no transaction hashes, decodes all transaction
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'ethereumtransactionsresource',
+        ), json={
+            'async_query': False,
+        },
+    )
+    assert_proper_response_with_result(response)
+    for tx_hash_hex in hashes:
+        receipt = dbethtx.get_receipt(hexstring_to_bytes(tx_hash_hex))
+        assert isinstance(receipt, EthereumTxReceipt) and receipt.tx_hash == hexstring_to_bytes(tx_hash_hex)  # noqa: E501
+        events = dbevents.get_history_events(
+            filter_query=HistoryEventFilterQuery.make(
+                event_identifier=tx_hash_hex,
+            ),
+            has_premium=True,  # for this function we don't limit. We only limit txs.
+        )
+        assert len(events) == 1
+        assert events[0].identifier not in event_ids
```

Comment:
So in this for loop you are basically copy pasting exactly what the previous test above does. Which ... does not test anything.

What do you want to test? That forcing requery, requeries all the events? Then you would need to mock the proper code parts so that you make sure that the code is entered again.

We do something similar here: https://github.com/rotki/rotki/blob/d8fc278229e7c3c4f2158f6168b276192c5d2e5f/rotkehlchen/tests/api/test_balances.py#L294

In fact I think that the previous function test where the coment `Check that force re-requesting the events works` needs to do it too, so you need to add the logic there too. And since it will probably end up being copy paste you would need to abstract it into a test function.